Mark_III_WB_Search/ViewController.mm

```diff
@@ -6,97 +6,44 @@
-
-    const NSInteger thisYear = [components valueForComponent:NSCalendarUnitYear];
-    components.calendar = calendar;
-
-    if (NULL != end) {
-        [components setValue:thisYear + plusYearsDelta - 1 forComponent:NSCalendarUnitYear];
-        [components setValue:12 forComponent:NSCalendarUnitMonth];
-        [components setValue:31 forComponent:NSCalendarUnitDay];
-        [components setValue:23 forComponent:NSCalendarUnitHour];
-        [components setValue:59 forComponent:NSCalendarUnitMinute];
-        [components setValue:59 forComponent:NSCalendarUnitSecond];
-
-        *end = components.date;
-    }
-
-    if (NULL != start) {
-        [components setValue:thisYear - minusYearsDelta forComponent:NSCalendarUnitYear];
-        [components setValue:1 forComponent:NSCalendarUnitMonth];
-        [components setValue:1 forComponent:NSCalendarUnitDay];
-        [components setValue:0 forComponent:NSCalendarUnitHour];
-        [components setValue:0 forComponent:NSCalendarUnitMinute];
-        [components setValue:0 forComponent:NSCalendarUnitSecond];
-
-        *start = components.date;
-    }
-}
-
-- (void)filterEventsInRange:(NSRange)range searchText:(NSString*)text resultIndexes:(NSMutableData*)resultIndexes {
-    if (0 == range.length) { return; }
-
-    [resultIndexes setLength:sizeof(NSInteger) * range.length];
-
-    NSUInteger index = 0;
-    NSInteger* indexesArray = (NSInteger*)resultIndexes.mutableBytes;
-
-    for (NSInteger idx = 0; idx < range.length; idx++) {
-        const NSInteger eventIdx = range.location + idx;
-
-        Event* event = self.events[eventIdx];
-        const NSRange substrRange = [event.title rangeOfString:text options:NSCaseInsensitiveSearch];
-        if (0 == substrRange.length) { continue; }
-
-        indexesArray[index] = eventIdx;
-        index++;
-    }
-
-    [resultIndexes setLength:sizeof(NSInteger) * index];
-}
 
 - (void)viewDidLoad {
     [super viewDidLoad];
 
     // init like
     self.eventCount = 10000; // TODO: predefined - change to something more useful
     self.filteredIndexesCount = -1;
+    self.highlightEventIndex = -1;
 
-    self.loadingQueue = dispatch_queue_create("com.viewController.loadingQueue", DISPATCH_QUEUE_SERIAL);
-    self.threadsCount = NSProcessInfo.processInfo.processorCount / 2;
+    _reportingQueue = dispatch_queue_create("com.viewController.loadingQueue", DISPATCH_QUEUE_SERIAL);
 
-    NSMutableArray* array = [NSMutableArray.alloc initWithCapacity:self.threadsCount];
-    for (NSUInteger idx = 0; idx < self.threadsCount; idx++) {
-        [array addObject:dispatch_queue_create([NSString stringWithFormat:@"com.viewController.processingQueues.%d", (int)idx].UTF8String, DISPATCH_QUEUE_SERIAL)];
+    const NSUInteger threadsCount = NSProcessInfo.processInfo.processorCount / 2;
```

Comment:
This solution unfortunately will work only for Intel, for Apple Silicone it will only slow down the search 
<img width="655" alt="Screenshot 2024-02-11 at 12 29 10" src="https://github.com/nesseratious/Mark_III_WB_Search/assets/36981242/ad75d9e7-ce74-41a9-b363-8dcd3a11f8d5">

Reply:
Useful information - many thanks! There are three classes that are currently only used for tests with other possible solutions. My plan for future improvements was to replace the current implementation with one of these.